### PR TITLE
Modules

### PR DESCRIPTION
**Pull request by Kaskyi**
https://github.com/xm-online/xm-webapp/pull/257
https://github.com/xm-online/xm-webapp/pull/259
https://github.com/xm-online/xm-webapp/pull/260
https://github.com/xm-online/xm-webapp/pull/261
https://github.com/xm-online/xm-webapp/pull/262
https://github.com/xm-online/xm-webapp/pull/263
https://github.com/xm-online/xm-webapp/pull/264
https://github.com/xm-online/xm-webapp/pull/265
https://github.com/xm-online/xm-webapp/pull/266
https://github.com/xm-online/xm-webapp/pull/267
https://github.com/xm-online/xm-webapp/pull/268
https://github.com/xm-online/xm-webapp/pull/269
https://github.com/xm-online/xm-webapp/pull/270
https://github.com/xm-online/xm-webapp/pull/271
https://github.com/xm-online/xm-webapp/pull/272
https://github.com/xm-online/xm-webapp/pull/273
#275
#276
#281

* add @xm-ngx/core/auth path;
* add @xm-ngx/shared modue;
* improve import of the sidebar styles;
* add XmSharedTestingModule;
* fix create test *.spec.ts;
* move JhiAlert to xm-alert.module.ts;
* AuthServerProvider  provide in root;
* split some shared components to the small modules: LoaderModule, XmMaintenanceViewModule, NoDataModule, XmPasswordNeededModule, LanguageModule, XmEntityModule.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ module.exports = {
         '@typescript-eslint/typedef': ['error', {arrayDestructuring: false, arrowParameter: false}],
         '@typescript-eslint/type-annotation-spacing': ['error'],
         '@typescript-eslint/explicit-member-accessibility': ['error', {
+            accessibility: 'off',
```
Comment (Sumragen) [inline .eslintrc.js]: Couldn't agree.

File: .eslintrc.js [hunk 2]
```diff
@@ -37,7 +38,7 @@ module.exports = {
         'line-comment-position': 'warn',
         'class-methods-use-this': 'warn',
         'no-plusplus': 'warn',
-        'indent': ['warn', 4, {'SwitchCase': 1}],
+        'indent': ['off', 4, {'SwitchCase': 1}],
```
Comment (Sumragen) [inline .eslintrc.js]: The same think as `explicit-member-accessibility`

File: src/app/modules/xm-shared/xm-shared-testing.module.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { RouterTestingModule } from '@angular/router/testing';
+import { TranslateService } from '@ngx-translate/core';
+import { of } from 'rxjs';
+
+import { XmSharedModule } from './xm-shared.module';
+
+@Injectable()
+@Pipe({name: 'translate'})
+export class TranslatePipe {
+    public transform: any = (arg) => of(arg);
```
Comment (Sumragen) [inline src/app/modules/xm-shared/xm-shared-testing.module.ts]: Avoid type `any` otherwise we could use JS

Comment (Kaskyi) [reply src/app/modules/xm-shared/xm-shared-testing.module.ts]: https://github.com/xm-online/xm-webapp/pull/281

File: src/app/shared/privilege/permission.directive.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    private elseTemplateRef: TemplateRef<PermissionContext> | null = null;
+    private thenViewRef: EmbeddedViewRef<PermissionContext> | null = null;
+    private elseViewRef: EmbeddedViewRef<PermissionContext> | null = null;
+
+    constructor(
+        private viewContainer: ViewContainerRef,
+        private permissionService: XmPermissionService,
+        templateRef: TemplateRef<PermissionContext>,
+    ) {
+        this.thenTemplateRef = templateRef;
```
Comment (didyk) [inline src/app/shared/privilege/permission.directive.ts]: Maybe can we transfer it to onInit?

File: src/app/shared/privilege/permission.directive.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        } else {
+            if (!this.elseViewRef) {
+                this.viewContainer.clear();
+                this.thenViewRef = null;
+                if (this.elseTemplateRef) {
+                    this.elseViewRef = this.viewContainer.createEmbeddedView(this.elseTemplateRef, this.context);
+                }
+            }
+        }
+    }
```
Comment (didyk) [inline src/app/shared/privilege/permission.directive.ts]: I don't like if in if in else. If we can maybe we rewrite it method?

File: browserslist [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+# https://github.com/browserslist/browserslist#queries
+
+# You can see what browsers were selected by your queries by running:
+#   npx browserslist
+
+> 0.5%
+last 2 versions
+Firefox ESR
+not dead
+IE 11 # For IE 9-11 support, remove 'not'.
```
[the remark this answers is not in the document]
Comment (Kaskyi) [reply browserslist]: Why not? Our application is able to support it

File: package.json [hunk 7]
```diff
@@ -24,7 +24,7 @@
     "dev": "npm run prebuild && ng serve --proxy-config local.proxy.conf.js --live-reload --host=0.0.0.0",
     "dev:aot": "npm run prebuild && ng serve --proxy-config local.proxy.conf.js --live-reload  --host=0.0.0.0 --prod --aot=true",
     "build": "npm run prebuild && ng build --stats-json",
-    "build-aot": "npm run prebuild && node --max_old_space_size=4096 ./node_modules/@angular/cli/bin/ng build --prod --aot=true",
+    "build-aot": "npm run prebuild && node --max_old_space_size=4096 ./node_modules/@angular/cli/bin/ng build --prod --aot=true --stats-json",
```
[the remark this answers is not in the document]
Comment (Kaskyi) [reply package.json]: to make a report about prod build

File: package.json [hunk 8]
```diff
@@ -33,6 +33,7 @@
     "prebuild": "npm run mng-ext && npm run install-exts && node ./scripts/merge-translations.js",
     "mng-ext": "node ./scripts/manage-exts.js",
     "test": "npm run prebuild && ng test --source-map=false",
+    "test:ci": "npm run prebuild && ng test --source-map=false --watch=false",
```
[the remark this answers is not in the document]
Comment (Kaskyi) [reply package.json]: to run tests with a pipeline,  `ci` to define the purpose for job

File: src/app/modules/xm-shared/xm-shared-testing.module.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+@Injectable()
+@Pipe({name: 'translate'})
+export class TranslatePipe {
+    public transform: (args: string) => Observable<string> = (arg) => of(arg);
+}
+
+export class TranslateServiceMock {
+    public get: (i: string) => Observable<string> = (arg) => of(arg);
+}
```
[the remark this answers is not in the document]
Comment (Kaskyi) [reply src/app/modules/xm-shared/xm-shared-testing.module.ts]: its xm-shared-testing.module.ts module for testing

File: browserslist (hunk 6 above)
[the remark this answers is not in the document]
Comment (Kaskyi) [reply browserslist]: also with angular 8 we supported it before

File: src/app/shared/privilege/permission.directive.ts (hunk 4 above)
Comment (Kaskyi) [reply src/app/shared/privilege/permission.directive.ts]: no its default value

File: src/app/shared/privilege/permission.directive.ts (hunk 5 above)
Comment (Kaskyi) [reply src/app/shared/privilege/permission.directive.ts]: I agree, it has many statements can u provide an example with alternative ?

File: src/app/shared/core/src/xm-ui-config.service.ts [hunk 10]
```diff
@@ -3,21 +3,33 @@ import { Inject, Injectable, OnDestroy } from '@angular/core';
...
 export class XmUiConfigService<T = XmUIConfig> implements OnDestroy {
 
-    protected requestCache: RequestCache<T> = new RequestCache<T>();
+    protected requestCache: RequestCache<T>;
 
     constructor(protected httpClient: HttpClient,
-                @Inject(XM_CORE_CONFIG) protected xmCoreConfig: XmCoreConfig,
+                @Inject(XmCoreConfig) protected xmCoreConfig: XmCoreConfig,
                 protected sessionService: XmSessionService) {
+        this.requestCache = new RequestCache<T>(() => this.publicAPI());
```
[the remark this answers is not in the document]
Comment (Kaskyi) [reply src/app/shared/core/src/xm-ui-config.service.ts]: done

File: src/app/shared/core/src/xm-user.service.ts [hunk 11]
```diff
@@ -2,21 +2,34 @@ import { HttpClient } from '@angular/common/http';
...
+        this.sessionService.get().pipe(
+            takeUntilOnDestroy(this),
+            filter<ISession>(Boolean),
+            pluck('active'),
+            distinctUntilChanged(),
+            map((isActive: boolean) => isActive ? this.getUser : (): Observable<null> => of(null)),
+        ).subscribe((request) => {
+            this.requestCache.request = request;
+            this.requestCache.forceReload();
+        });
```
[the remark this answers is not in the document]
Comment (Kaskyi) [reply src/app/shared/core/src/xm-user.service.ts]: done

**Pull request merged**